docs/scripts/api/buildInterfacesDocumentation.ts

```diff
@@ -173,123 +177,55 @@ const parseInterfaceSymbol = async (
   return parsedInterface;
 };
 
-function generateMarkdownFromProperties(
-  object: ParsedObject,
-  documentedInterfaces: DocumentedInterfaces,
-) {
-  const hasDefaultValue = object.properties.some((property) => {
-    return property.tags.default;
-  });
-
-  const headers = hasDefaultValue
-    ? `
-| Name | Type | Default | Description |
-|:-----|:-----|:--------|:------------|`
-    : `
-| Name | Type | Description |
-|:-----|:-----|:------------|`;
-
-  let text = `${headers}\n`;
-
-  object.properties.forEach((property) => {
-    const defaultValue = property.tags.default?.text?.[0].text;
-
-    let planImg: string;
-    if (property.projects.includes('x-data-grid')) {
-      planImg = '';
-    } else if (property.projects.includes('x-data-grid-pro')) {
-      planImg =
-        ' [<span class="plan-pro" title="Pro plan"></span>](/x/introduction/licensing/#pro-plan)';
-    } else if (property.projects.includes('x-data-grid-premium')) {
-      planImg =
-        ' [<span class="plan-premium" title="Premium plan"></span>](/x/introduction/licensing/#premium-plan)';
-    } else {
-      throw new Error(`No valid plan found for ${property.name} property in ${object.name}`);
-    }
-
-    const formattedName = property.isOptional
-      ? `<span class="prop-name optional">${property.name}<sup><abbr title="optional">?</abbr></sup>${planImg}</span>`
-      : `<span class="prop-name">${property.name}${planImg}</span>`;
-
-    const formattedType = `<span class="prop-type">${escapeCell(property.typeStr)}</span>`;
-
-    const formattedDefaultValue =
-      defaultValue == null ? '' : `<span class="prop-default">${escapeCell(defaultValue)}</span>`;
-
-    const formattedDescription = escapeCell(
-      linkify(property.description, documentedInterfaces, 'markdown'),
-    );
-
-    if (hasDefaultValue) {
-      text += `| ${formattedName} | ${formattedType} | ${formattedDefaultValue} | ${formattedDescription} |\n`;
-    } else {
-      text += `| ${formattedName} | ${formattedType} | ${formattedDescription} |\n`;
-    }
-  });
```

Comment:
Formatting and linkify is done when generating the JSON. The rendering aspect (column order, adding plan image, ...) is done in the `PropsTable` component done in https://github.com/mui/material-ui/pull/41069